### PR TITLE
mirror my version of the algo more closely

### PR DESCRIPTION
**Pull request by nacrt**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -90,4 +90,4 @@ shadowJar {
     exclude "META-INF/versions/**"
 }
 
-tasks.reobfJar.dependsOn(shadowJar)
```
Comment (My-Name-Is-Jeff) [inline build.gradle:93]: what

File: src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt [hunk 2]
```diff
@@ -163,8 +163,8 @@ object UpdateChecker {
 
         val e = expected.toCharArray().dropWhile { it == '!' }.filter { !exempt.contains(it) }
         val r = received.toCharArray().dropWhile { it == '!' }.filter { !exempt.contains(it) }
-        val distance = StringUtils.getLevenshteinDistance(expected, received)
-        if (distance !in 1..6) return false
+        val distance = StringUtils.getLevenshteinDistance(e, r)
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt]: this is 2 char arrays, not 2 strings

you need to convert them to strings first

**Review by My-Name-Is-Jeff — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (nacrt) [reply build.gradle:93]: git moment

**Comment by nacrt** [comment]
done

File: src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt [hunk 3]
```diff
@@ -163,8 +163,8 @@ object UpdateChecker {
 
         val e = expected.toCharArray().dropWhile { it == '!' }.filter { !exempt.contains(it) }
         val r = received.toCharArray().dropWhile { it == '!' }.filter { !exempt.contains(it) }
-        val distance = StringUtils.getLevenshteinDistance(expected, received)
-        if (distance !in 1..6) return false
+        val distance = StringUtils.getLevenshteinDistance(e.toString(), r.toString())
```
Comment (My-Name-Is-Jeff) [inline src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt]: thats not how it works

**Review by My-Name-Is-Jeff — CHANGES_REQUESTED** [review]

File: src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt (hunk 3 above)
Comment (My-Name-Is-Jeff) [reply src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/UpdateChecker.kt]: joinToString

**Pull request merged**